### PR TITLE
Save Data linux support

### PR DESCRIPTION
**Pull request by spham-amzn**
* Implement the SaveData gem for Linux
* Enable unit tests for SaveData gem for Linux

### CODE REVIEW

File: Gems/SaveData/Code/Source/Platform/Linux/SaveData_SystemComponent_Linux.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+#include <sys/types.h>
+#include <pwd.h>
+#include <unistd.h>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+namespace SaveData
+{
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    //! Platform specific implementation for the save data system component on Windows
```
Comment (bosnichd) [inline Gems/SaveData/Code/Source/Platform/Linux/SaveData_SystemComponent_Linux.cpp:25]: on Windows -> on Linux

File: Gems/SaveData/Code/Source/Platform/Linux/SaveData_SystemComponent_Linux.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        //! \ref SaveData::SaveDataSystemComponent::Implementation::SetSaveDataDirectoryPath
+        void SetSaveDataDirectoryPath(const char* saveDataDirectoryPath) override;
+
+    private:
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Convenience function to construct the full save data file path.
+        //! \param[in] dataBufferName The name of the save data buffer.
+        //! \param[in] localUserId The local user id the save data buffer is associated with.
+        AZ::IO::Path GetSaveDataFilePath(const AZStd::string& dataBufferName,
+                                              AzFramework::LocalUserId localUserId);
```
Comment (bosnichd) [inline Gems/SaveData/Code/Source/Platform/Linux/SaveData_SystemComponent_Linux.cpp:64]: Nit-pick: indentation looks off? Might just be the browser display.

File: Gems/SaveData/Code/Source/Platform/Linux/SaveData_SystemComponent_Linux.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            homeDir = pwuid->pw_dir;
+        }
+
+        AZ_Assert(homeDir, "Unable to determine home directory for current Linux user");
+        if (homeDir == nullptr)
+        {
+            homeDir = "/tmp";
+        }
+
+        // AZ::IO::PathView homePath {homeDir};
```
Comment (bosnichd) [inline Gems/SaveData/Code/Source/Platform/Linux/SaveData_SystemComponent_Linux.cpp:91]: Remove comment?

**Review by bosnichd — APPROVED** [review]
Great, thanks!

**Review by AMZN-ScottR — APPROVED** [review]

**Review by bosnichd — APPROVED** [review]

**Pull request merged**